Tools/LyTestTools/ly_test_tools/o3de/editor_test.py

```diff
@@ -536,7 +553,8 @@ def single_run(self, request, workspace, editor, editor_test_data, launcher_plat
                         if is_single_test:
                             # Setup step for wrap_run
                             wrap = inner_test_spec.wrap_run(self, request, workspace, editor, editor_test_data, launcher_platform)
-                            assert isinstance(wrap, types.GeneratorType), "wrap_run must return a generator, did you forget 'yield'?"
+                            if not isinstance(wrap, types.GeneratorType):
+                                raise exceptions.LyTestToolsException("wrap_run must return a generator, did you forget 'yield'?")
```

Comment:
Here and throughout editor_test.py, should these exceptions not be EditorTestExceptions?

Reply:
I think an EditorTestFrameworkException/Error could make it clearer. Could also suggest naming our other new one LyTestToolsFrameworkException/Error